cloudpickle/cloudpickle_fast.py

```diff
@@ -0,0 +1,618 @@
+"""
+New, fast version of the Cloudpickler.
+
+This new Cloudpickler class can now extend the fast C Pickler instead of the
+previous pythonic Pickler. Because this functionality is only available for
+python versions 3.8+, a lot of backward-compatibilty code is also removed.
+"""
+import abc
+import io
+import itertools
+import logging
+import _pickle
+import pickle
+import sys
+import types
+import weakref
+
+from _pickle import Pickler
+
+from .cloudpickle import (
+    islambda, _is_dynamic, extract_code_globals, _BUILTIN_TYPE_CONSTRUCTORS,
+    _BUILTIN_TYPE_NAMES, DEFAULT_PROTOCOL, _find_loaded_submodules,
+    _get_cell_contents
+)
+
+load, loads = _pickle.load, _pickle.loads
+
+# Shorthands similar to pickle.dump/pickle.dumps
+
+
+def dump(obj, file, protocol=None):
+    """Serialize obj as bytes streamed into file
+
+    protocol defaults to cloudpickle.DEFAULT_PROTOCOL which is an alias to
+    pickle.HIGHEST_PROTOCOL. This setting favors maximum communication speed
+    between processes running the same Python version.
+
+    Set protocol=pickle.DEFAULT_PROTOCOL instead if you need to ensure
+    compatibility with older versions of Python.
+    """
+    CloudPickler(file, protocol=protocol).dump(obj)
+
+
+def dumps(obj, protocol=None):
+    """Serialize obj as a string of bytes allocated in memory
+
+    protocol defaults to cloudpickle.DEFAULT_PROTOCOL which is an alias to
+    pickle.HIGHEST_PROTOCOL. This setting favors maximum communication speed
+    between processes running the same Python version.
+
+    Set protocol=pickle.DEFAULT_PROTOCOL instead if you need to ensure
+    compatibility with older versions of Python.
+    """
+    file = io.BytesIO()
+    try:
+        cp = CloudPickler(file, protocol=protocol)
+        cp.dump(obj)
+        return file.getvalue()
+    finally:
+        file.close()
```

Comment:
This could be written as:

```python
with io.BytesIO() as f:
    cp = CloudPickler(file, protocol=protocol)
    cp.dump(obj)
    return f.getvalue()
```